common/settings/settings.ts

```diff
@@ -30,6 +30,96 @@ export interface MiscSettings {
     readonly pauseOnHoverMode: PauseOnHoverMode;
 }
 
+/*
+These are all the possible scenarios which can result in a match. We don't need to support every possible combination,
+as some are not useful or inconsistent. Inconsistent meaning the order the user collects forms affects what future forms
+are considered collected (e.g collecting the lemma will match all forms but user needs to collect every inflection if
+they don't ever collect the lemma).
+
+Lemma In Subtitle
+-----------------------------------------------------------------
+| User Collection | LEMMA_FORM_COLLECTED | EXACT_FORM_COLLECTED |
+-----------------------------------------------------------------
+| Lemma           |         MATCH        |         MATCH        |
+| Inflection      |          NO          |          NO          |
+-----------------------------------------------------------------
+
+Inflection In Subtitle
+-----------------------------------------------------------------
+| User Collection | LEMMA_FORM_COLLECTED | EXACT_FORM_COLLECTED |
+-----------------------------------------------------------------
+| Lemma           |         MATCH        |          NO          |
+| Same Inflection |          NO          |         MATCH        |
+| Diff Inflection |          NO          |          NO          |
+-----------------------------------------------------------------
+*/
+export enum TokenMatchStrategy {
+    ANY_FORM_COLLECTED = 'ANY_FORM_COLLECTED', // All scenarios above result in MATCH
+    LEMMA_OR_EXACT_FORM_COLLECTED = 'LEMMA_OR_EXACT_FORM_COLLECTED', // See LEMMA_FORM_COLLECTED and EXACT_FORM_COLLECTED columns above
+    LEMMA_FORM_COLLECTED = 'LEMMA_FORM_COLLECTED', // See LEMMA_FORM_COLLECTED column above
+    EXACT_FORM_COLLECTED = 'EXACT_FORM_COLLECTED', // See EXACT_FORM_COLLECTED column above
+}
+
+export enum TokenMatchStrategyPriority {
+    EXACT = 'EXACT',
+    LEMMA = 'LEMMA',
+    BEST_KNOWN = 'BEST_KNOWN',
+    LEAST_KNOWN = 'LEAST_KNOWN',
+}
+
+export enum DictionaryAnkiTreatSuspended {
+    NORMAL = 'normal',
+    MATURE = 'mature',
+    YOUNG = 'young',
+    UNKNOWN = 'unknown',
+}
+
+export enum TokenStyling {
+    TEXT = 'text',
+    UNDERLINE = 'underline',
+    OVERLINE = 'overline',
+}
+
+export enum TokenStatus {
+    UNCOLLECTED = 0,
+    UNKNOWN = 1,
+    YOUNG = 2,
+    MATURE = 3, // If ever adding more statues, they should go after MATURE and getFullyKnownTokenStatus should be updated
+}
+
+export function getFullyKnownTokenStatus(): TokenStatus {
+    return TokenStatus.MATURE; // If future statuses are optional, this logic may need to change
+}
+
+export interface DictionarySubtitleAppearance {
+    readonly tokenStyling: TokenStyling;
+    readonly tokenStylingThickness: number;
+    readonly colorizeFullyKnownTokens: boolean;
+    readonly tokenStatusColors: string[]; // Indexed by TokenStatus
+}
+
+export interface DictionaryTrack {
+    readonly dictionaryColorizeOnVideo: boolean;
```

Comment:
I understand the desire for flexibility, but this also seems like a setting that doesn't need app/extension-specific variants. App/extension settings were unified with 1.0 so I'd prefer if we only go in the opposite direction only if there's an obvious use-case to do so. In this case it doesn't seem obvious to me but let me know if you think otherwise.

Reply:
I think a display toggle could be nice. Maybe not for this PR. Yes, re-rendering the subtitle lists could be expensive.. but we already have a similar feature that simply hides the text. Hopefully users don't spam the button.